solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -2675,7 +2676,12 @@ public void execute(SolrRequestHandler handler, SolrQueryRequest req, SolrQueryR
 
     if (rsp.getToLog().size() > 0) {
       if (requestLog.isInfoEnabled()) {
-        requestLog.info(rsp.getToLogAsString(logid));
+        Object path = rsp.getToLog().get("path");
+        if (path instanceof String) {
```

Comment:
nit: may be check `path != null` and then use `path.toString()`.

Reply:
Thanks Shan! 

I skipped this, as `null instanceof String` is always false, hence after this point path should be a non null string :)

Reply:
Yeah, I mean use null check to replace instanceof, if your intention is mostly for checking and use path as a string. instanceof may be slow. But it is fine to keep it as is.